ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/Praos/Common.hs

```diff
@@ -104,6 +104,12 @@ comparePraos tiebreakerFlavor =
     when' p comp a1 a2 =
         if p a1 a2 then comp a1 a2 else EQ
 
+    -- Only compare the issue numbers when the issuers and slots are identical.
+    -- Note that in this case, the VRFs must also coincide.
+    issueNoArmed v1 v2 =
+           csvSlotNo v1 == csvSlotNo v2
+        && csvIssuer v1 == csvIssuer v2
```

Comment:
As mentioned in the PR description, we could also replace this with ``(==) `on` csvTieBreakVRF``, and even remove the `csvIssuer` from `PraosChainSelectView`. This would probably be a bit faster (might be irrelevant), but also slightly less clear.

Reply:
We'd ultimately need empiricism here (if we suspect this is not irrelevant), but it also seems plausible that the existing check would be _faster_, since slot comparisons are cheaper, and I suspect most height battles are _not_ slot battles 🤔

Reply:
Right, then I will leave it as is :+1: